libs/designer-ui/src/lib/authentication/plugins/CollapsedAuthenticationSerialization.tsx

```diff
@@ -0,0 +1,86 @@
+import type { AuthProps } from '..';
+import { AuthenticationType } from '..';
+import type { ValueSegment } from '../../editor';
+import { serializeEditorState } from '../../editor/base/utils/editorToSegment';
+import { getChildrenNodes } from '../../editor/base/utils/helper';
+import { serializeAuthentication, validateAuthenticationString } from '../util';
+import { css } from '@fluentui/react';
+import { OnChangePlugin } from '@lexical/react/LexicalOnChangePlugin';
+import { isTemplateExpression } from '@microsoft/parsers-logic-apps';
+import type { EditorState } from 'lexical';
+import { $getRoot } from 'lexical';
+import type { Dispatch, SetStateAction } from 'react';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+
+export interface CollapsedAuthenticationValidationProps {
+  className?: string;
+  setToggleEnabled: (b: boolean) => void;
+  setCurrentProps: Dispatch<SetStateAction<AuthProps>>;
+  setOption: (s: AuthenticationType) => void;
+  serializeValue: (value: ValueSegment[]) => void;
+}
+
+export const CollapsedAuthenticationSerialization = ({
+  className,
+  setToggleEnabled,
+  setCurrentProps,
+  setOption,
+  serializeValue,
+}: CollapsedAuthenticationValidationProps): JSX.Element => {
+  const intl = useIntl();
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const showError = (errorMessage: string) => {
+    setToggleEnabled(false);
```

Comment:
instead of juggling multiple stages for this, couldn't you just infer toggleEnabled from the errorMessage being set?